### PR TITLE
Feat/native/review and send

### LINKED ISSUE #10867: Send coins - inputs review screen
## Description
After filling an input form, user should be able to validate these values on the device screen.

## Acceptance criteria
- [x] After submiting a valid send form user is redirected to another screen where he can review filled inputs
- [x] After user goes through the device process, the screen renders simple message saying that verification was successful

### LINKED ISSUE #10868: Send coins - send transaction
## Description
After validating the send values by device, user should be able to create transaction and send it.

## Acceptance criteria
- [ ] after the device values verification, user is able to click `Send` button
- [ ] button click triggers on device send flow
- [ ] after successful device send flow, the user is in app notified that send was successful and redirected back to the account screen

### PR DESCRIPTION
**Pull request by PeKne**
Enables basic send form submission of BTC testnets (TEST and REGTEST).  While reviewing this, please focus on the design of the whole flow and code architecture. Reviewing UI or some styling/translation nits is not necessary at this stage, because this implementation does not reflect the final design. The purpose of this PR is solely to prepare the codebase so it can send the transaction. 

Also there are some steps of the send flow that are not implemented yet but will be revisited in follow up PRs (eg. fee selection, multiple recipients...)

## Description
The PR is divided into two commits:
- [5d693e5](https://github.com/trezor/trezor-suite/pull/12968/commits/5d693e5119f6ee36bdaf3184523b6be4766d113a)- **moves transaction outputs utils from `packages` to `suite-common` to be reused in native**. There are no new things implemented only moving stuff around.
- [f6df275](https://github.com/trezor/trezor-suite/pull/12968/commits/f6df275bfd83f9f9696708e44d2bc787b6780d67) -  **transaction review & send**
  - Review screen reflecting on-device confirmation steps added
  - basic error handling implemented
  - on successful on-device verification, the user can click the `send transaction` button and broadcast the transaction to the blockchain
  - note: sorry for making this PR so big, but most of it is just UI so I hope it's not that big a deal. In case you need a better explanation of this PR, ping me on Slack
 
## Related Issues

Resolve #10867
Resolve #10868

## Screenshots:

https://github.com/trezor/trezor-suite/assets/26143964/6757c098-fa24-406f-b7dd-1a0df8acb83a

### CODE REVIEW

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx [hunk 1]
```diff
@@ -60,25 +63,28 @@ export const TransactionReviewModalContent = ({
     const { account } = selectedAccount;
     const { precomposedTx, serializedTx } = txInfoState;
 
+    const precomposedForm: FormState | StakeFormState | undefined = isStakeState(txInfoState)
+        ? txInfoState.precomposedForm
+        : txInfoState.drafts[account?.key ?? ''];
+
+    const decreaseOutputId = precomposedTx?.useNativeRbf
+        ? precomposedForm?.setMaxOutputId
+        : undefined;
```
Comment (juriczech) [inline packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx]: Curious: Why was this moved above the check of non-null account? Now instead of `account.key`, you are doing `account?.key ?? ""` with no added benefit? If it's needed for the selector, there should be a selector for this for the logic to be hidden there. It's a small thing, but I think you should create a selector for this to hide this logic there.

File: suite-common/wallet-types/src/transaction.ts [hunk 2]
```diff
@@ -243,3 +243,41 @@ export type TransactionFiatRateUpdatePayload = {
 export type TransactionType = Pick<WalletAccountTransaction, 'type'>['type'];
 
 export type ExportFileType = 'csv' | 'pdf' | 'json';
+
+export type ReviewOutput =
```
Comment (juriczech) [inline suite-common/wallet-types/src/transaction.ts:247]: I would move this to send package as explained in earlier comment

File: suite-common/wallet-utils/src/reviewTransactionUtils.ts [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    FormState,
+    GeneralPrecomposedTransactionFinal,
+    ReviewOutputState,
+} from '@suite-common/wallet-types';
+import { StakeFormState, ReviewOutput } from '@suite-common/wallet-types';
+import { TrezorDevice } from '@suite-common/suite-types';
+
+import { getShortFingerprint, isCardanoTx } from './cardanoUtils';
+
+export const getOutputState = (index: number, buttonRequestsCount: number): ReviewOutputState => {
```
Comment (juriczech) [inline suite-common/wallet-utils/src/reviewTransactionUtils.ts]: Similar here: It should be in send package and I'd also consider better naming for these functions. `getOutputState` in regards to wallet-utils is not really explanatory enough.

File: packages/suite/src/types/wallet/transaction.ts [hunk 4]
```diff
@@ -1,39 +1,3 @@
...
-      }
-    | {
-          type: 'reduce-output';
-          label: string;
-          value: string;
-          value2: string;
-          token?: undefined;
-      };
-
 export type {
```
Comment (juriczech) [inline packages/suite/src/types/wallet/transaction.ts]: IMO this file no longer serves it's purpose. You've stripped the only reason there was to leave this file as is and now it's just re-exporting other types. Wouldn't it be better to just move the whole file and rename it? Git might treat it as just moving + renaming since changes would be minimal (removing re-export) and it could still be shared code.

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 5]
```diff
@@ -923,3 +924,37 @@ export const selectInstacelessUnselectedDevices = memoize((state: DeviceRootStat
 
     return deviceUtils.getSortedDevicesWithoutInstances(allDevices, device?.id);
 });
+
+export const selectSendFormReviewButtonRequestsCount = (
```
Comment (juriczech) [inline suite-common/wallet-core/src/device/deviceReducer.ts]: This could be added to send form reducer maybe? Is there a reason why you couldn't import `DeviceRootState` or some selectors from this package in there? IMO it would be better.

File: packages/suite/src/types/wallet/transaction.ts (hunk 4 above)
Comment (juriczech) [inline packages/suite/src/types/wallet/transaction.ts]: This file no longer has any purpose. Please delete it because now it's just re-exporting types which shouldn't happen / shouldn' be needed.

File: suite-native/module-send/src/components/ReviewOutputItem.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    return (
+        <Text numberOfLines={1} adjustsFontSizeToFit>
+            {value}
+        </Text>
+    );
+};
+
+// TODO: adjust UI to the design when ready
```
Comment (juriczech) [inline suite-native/module-send/src/components/ReviewOutputItem.tsx]: I know it's nitpick, but please let's not add this type of comments everywhere. This really is redundant. If it's in some design and issue, it will be done and without linking PR it's just clutter.

File: suite-native/module-send/src/components/ReviewOutputItem.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+export const ReviewOutputItem = ({
+    reviewOutput,
+    networkSymbol,
+}: {
+    networkSymbol: NetworkSymbol;
+    reviewOutput: StatefulReviewOutput;
+    status: ReviewOutputState;
+}) => {
```
Comment (juriczech) [inline suite-native/module-send/src/components/ReviewOutputItem.tsx]: `ReviewOutputItemProps` at the top of the file please

File: suite-native/module-send/src/components/ReviewOutputItem.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    switch (state) {
+        case 'active':
+            return 'warning';
+        case 'success':
+            return 'success';
+        default:
+            return 'error';
+    }
+};
+
```
Comment (juriczech) [inline suite-native/module-send/src/components/ReviewOutputItem.tsx]:
```
const alertBoxVariantMap = {
    active: "warning",
    success: "success",
} as const satisfies Record<ReviewOutputState, AlertBoxVariant>
```

Then use in component could look like `alertBoxVariantMap[output.state] ?? 'error'`

This seems better and less verbose to me.

File: suite-native/module-send/src/components/SendOutputFields.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+const amountTransformer = (value: string) =>
+    value
+        .replace(/[^0-9\.]/g, '') // remove all non-numeric characters
+        .replace(/^\./g, '') // remove '.' symbol if it is not preceded by number
+        .replace(/(?<=\..*)\./g, '') // keep only first appearance of the '.' symbol
+        .replace(/(?<=^0+)0/g, ''); // remove all leading zeros except the first one
+
+export const SendOutputFields = () => {
+    const { control } = useContext(FormContext);
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputFields.tsx]: Why aren't you using `useFormContext`?

File: suite-native/module-send/src/components/SendOutputsForm.tsx [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+import { SendOutputFields } from './SendOutputFields';
+
+type SendFormProps = {
+    accountKey: AccountKey;
+};
+
+// TODO: this data structure will be revisited in a follow up PR
+const constructFormDraft = ({ outputs }: SendOutputsFormValues): FormState => ({
+    outputs:
+        outputs?.map(({ address, amount }) => ({
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputsForm.tsx]: Isn't `outputs.map()` sufficient since it's array and if it's empty array, you don't need to do that?

File: suite-native/module-send/src/components/SendOutputsForm.tsx [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+    const {
+        handleSubmit,
+        control,
+        getValues,
+        formState: { isValid, isSubmitting },
+    } = form;
+    const watchedFormValues = useWatch({ control });
+
+    const storeFormDraftIfValid = useCallback(async () => {
+        await dispatch(
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputsForm.tsx]: Redundant async / await?

File: suite-native/module-send/src/components/SendOutputsForm.tsx [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+        },
+    });
+
+    const {
+        handleSubmit,
+        control,
+        getValues,
+        formState: { isValid, isSubmitting },
+    } = form;
+    const watchedFormValues = useWatch({ control });
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputsForm.tsx:94]: No need to pass `control`, since it's optional when using form provider.

File: suite-native/module-send/src/components/SendOutputsForm.tsx [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+        );
+    }, [accountKey, dispatch, getValues]);
+
+    useEffect(() => {
+        if (watchedFormValues && isValid) debounce(storeFormDraftIfValid);
+    }, [isValid, storeFormDraftIfValid, watchedFormValues, debounce]);
+
+    const handleNavigateToReviewScreen = handleSubmit(async () => {
+        // Ignore another submit button clicks until is the submit not finished.
+        if (isSubmitting) return;
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputsForm.tsx]: You should disable the button if `isSubmitting` is true.

File: suite-native/module-send/src/components/SendOutputsForm.tsx [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+
+        // TODO: navigate to SendFeeScreen instead, when ready
+        // issue: https://github.com/trezor/trezor-suite/issues/10871
+        navigation.navigate(SendStackRoutes.SendReview, { accountKey });
+        const response = await dispatch(
+            onDeviceTransactionReviewThunk({
+                accountKey,
+                formState: constructFormDraft(getValues()),
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputsForm.tsx]: You don't need to `getValues()`, `handleSubmit` passes these as a parameter to the anonymous function so they can be used here.

File: suite-native/module-send/src/components/SendOutputsForm.tsx [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+            }),
+        );
+
+        if (isRejected(response)) {
+            // TODO: display error message based on the error code
+            showToast({ variant: 'error', message: 'Something went wrong', icon: 'closeCircle' });
+
+            navigation.navigate(SendStackRoutes.SendAccounts);
```
Comment (juriczech) [inline suite-native/module-send/src/components/SendOutputsForm.tsx:124]: I believe this should be `navigation.goBack`. We should count with send module being opened from multiple places and if we navigate back, we might want to go back where user came from instead of specific screen (we do that in a lot of other places)

File: suite-native/module-send/src/screens/SendOutputsScreen.tsx [hunk 16]
```diff
@@ -45,7 +45,7 @@ export const SendFormScreen = ({
                         />
                     </HStack>
                 </VStack>
-                <SendForm accountKey={accountKey} />
+                <SendOutputsForm accountKey={accountKey} />
```
Comment (juriczech) [inline suite-native/module-send/src/screens/SendOutputsScreen.tsx:57]: I believe it should either be as it was. We're not sending outputs per say, we're just sending transaction which has outputs, no? This name is pretty confusing IMO

File: suite-native/module-send/src/screens/SendReviewScreen.tsx [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+        if (isRejected(sendResponse)) {
+            // TODO: error handling
+        }
+
+        showToast({ variant: 'success', message: 'Transaction sent', icon: 'check' });
+
+        navigation.dispatch(navigateToAccountDetail({ accountKey }));
+    };
+
+    const handleGoBack = () => {
```
Comment (juriczech) [inline suite-native/module-send/src/screens/SendReviewScreen.tsx]: `handleCancel` might be better name? Is there an actual redirect happening?

File: suite-native/module-send/src/screens/SendReviewScreen.tsx [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+    const signedTransaction = useSelector(selectSendSignedTx);
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, accountKey),
+    );
+
+    if (!account) return;
+
+    const handleSendTransaction = async () => {
+        // TODO: error handling
+        if (!signedTransaction) return;
```
Comment (juriczech) [inline suite-native/module-send/src/screens/SendReviewScreen.tsx]: I believe this should be in outer scope (similar to account condition) to return null (or better yet - ErrorMessage from atoms) if no transaction was found by selector? This function should be called just when there is a transaction to be sent IMO and we just handle reject from sending it.

File: suite-native/navigation/src/components/GoBackIcon.tsx [hunk 19]
```diff
@@ -5,10 +5,11 @@ import { IconButton } from '@suite-native/atoms';
 import { CloseActionType } from '../navigators';
 
 type GoBackIconProps = {
+    onPress?: () => void;
```
Comment (juriczech) [inline suite-native/navigation/src/components/GoBackIcon.tsx]: I would advise against this if you're sending onPress which doesn't go back, then it shouldn't be in GoBack component :D

**Review by juriczech — CHANGES_REQUESTED** [review]
Overall good job! Just some things I'd rather get right from the beginning rather than creating code that will be refactored. I'm sorry if this sounds the wrong way, but IMO in the future, we shouldn't create PR this size with saying most of it will change. If I'm reviewing some code, I'd like to get it right to build on top of it or don't do it in a PR like this. Becuase then it might never be changed and if we don't review it then it's a recipe for unmaintainability. However it's just some things for discussion so let's see if they're important or not. Feel free to huddle me if you want to further discuss them.

**Comment by juriczech** [comment]
This pull request can be closed as redundant now? https://github.com/trezor/trezor-suite/pull/12240

File: packages/suite/src/types/wallet/transaction.ts (hunk 4 above)
Comment (PeKne) [reply packages/suite/src/types/wallet/transaction.ts]: duplicate comment, closing.

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx (hunk 1 above)
Comment (PeKne) [reply packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx]: The thing is that this cannot be moved to selector in context of this file because of the way how is this component designed. The whole state of this module is served a as a component prop (yes, you heard me right 😩) and it can be either `SendState` or `StakeState`. These states are very similar, but not the same 😤. So if I would create a selector in `SendReducer` that we use also in mobile, it would broke the desktop send form for staking. 

I agree that it is not ideal, but I'm sure I don't want to refactor this as part of this PR. Because it would be huge refactor across many packages of desktop suite.

File: packages/suite/src/types/wallet/transaction.ts (hunk 4 above)
Comment (PeKne) [reply packages/suite/src/types/wallet/transaction.ts]: fixed in a883ca4bae. I already remaped the imports in the original PR, i just forgot to remove this leftover 🙈.

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 5 above)
Comment (PeKne) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: moved a883ca4bae

File: suite-native/module-send/src/components/ReviewOutputItem.tsx (hunk 6 above)
Comment (PeKne) [reply suite-native/module-send/src/components/ReviewOutputItem.tsx]: I removed it. But you would be suprised that I got some comment in the older PR, asking me if it is the final solution :D. For this reason I included this comment.

File: suite-common/wallet-utils/src/reviewTransactionUtils.ts (hunk 3 above)
Comment (PeKne) [reply suite-common/wallet-utils/src/reviewTransactionUtils.ts]: Named changed for better context. a883ca4bae

File: suite-native/module-send/src/components/ReviewOutputItem.tsx (hunk 7 above)
Comment (PeKne) [reply suite-native/module-send/src/components/ReviewOutputItem.tsx]: fixed 3fd9ef1399

File: suite-native/module-send/src/components/ReviewOutputItem.tsx (hunk 8 above)
Comment (PeKne) [reply suite-native/module-send/src/components/ReviewOutputItem.tsx]: fixed 3fd9ef1399

File: suite-native/module-send/src/components/SendOutputFields.tsx (hunk 9 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputFields.tsx]: I wasn't aware we have it 👍 , fixed: 3fd9ef1399

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 10 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: fixed by making the otputs required in yup validation schema: 3fd9ef1399

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 12 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx:94]: This hook is not called inside of the form context (the `Form` is defined in this file), For this reason, the `control` argument is needed.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 13 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: fixed 3fd9ef1399

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 14 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: good point, fixed 3fd9ef1399

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 15 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx:124]: The information from design is that it should behave this way. If user clicks this button, he goes back to the send account selection.

File: suite-native/module-send/src/screens/SendOutputsScreen.tsx (hunk 16 above)
Comment (PeKne) [reply suite-native/module-send/src/screens/SendOutputsScreen.tsx:57]: It has to be like this. Because in the next PR will be added `SendFeesForm`.

File: suite-native/module-send/src/screens/SendReviewScreen.tsx (hunk 17 above)
Comment (PeKne) [reply suite-native/module-send/src/screens/SendReviewScreen.tsx]: renamed 3fd9ef1399

File: suite-native/navigation/src/components/GoBackIcon.tsx (hunk 19 above)
Comment (PeKne) [reply suite-native/navigation/src/components/GoBackIcon.tsx]: You are right. The GoBackIcon changes reverted: 3fd9ef1399

File: suite-native/module-send/src/screens/SendReviewScreen.tsx (hunk 18 above)
Comment (PeKne) [reply suite-native/module-send/src/screens/SendReviewScreen.tsx]: Most of the lifecycle of this component, there should be no `signedTransaction`. The component first show the review steps and only if user confirms all of them, the signed transaction is created. For this reason, it can not be handled in out of this function.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 11 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: good catch. Fixed by: 3b8cbb289f. I had to change typing of `useAsyncDebounce` to allow to send simple synchronous function as a argument.

Comment (juriczech) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: I woul also rename the hook to `useDebounce` as it doesn't only accept async functions now. Or maybe separate into 2 different functions. However I myself prefer the first option.

File: suite-native/navigation/src/components/GoBackIcon.tsx (hunk 19 above)
Comment (juriczech) [reply suite-native/navigation/src/components/GoBackIcon.tsx]: But you're still not doing any navigation. Am I missing something?

File: suite-native/module-send/src/screens/SendReviewScreen.tsx (hunk 18 above)
Comment (juriczech) [reply suite-native/module-send/src/screens/SendReviewScreen.tsx]: But the button that triggers this function already checks that `signedTransaction` is present. So I think you can just remove this.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 15 above)
Comment (juriczech) [reply suite-native/module-send/src/components/SendOutputsForm.tsx:124]: Correct me if I'm wrong, but you're only navigating to this screen from SendAccounts - meaning navigation go back would do the same thing. 

Benefit of using `navigation.goBack` is that it would also work if we add new route from which you can navigate to send form. If user clicks back in that case, we would want him to be redirected back, not to send accounts.

File: suite-common/wallet-utils/src/reviewTransactionUtils.ts [hunk 20]
```diff
@@ -0,0 +1,336 @@
...
+};
+
+type ConstructOutputsParams = {
+    precomposedTx: GeneralPrecomposedTransactionFinal;
+    decreaseOutputId: number | undefined;
+    account: Account;
+    precomposedForm: FormState | StakeFormState;
+};
+
+const constructOldFlow = ({
```
Comment (juriczech) [inline suite-common/wallet-utils/src/reviewTransactionUtils.ts:88]: Can you please add comments to this old vs new flow? What does that even mean? 😆 Rename these please, if possible.

**Review by Nodonisko — COMMENTED** [review]
Seems that @juriczech already pointed out most of issues, only one thing that I have in mind is that we should call `updateFeeInfoThunk` in the begging of send flow because on mobile we don't update fee with every block like on desktop.

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx (hunk 1 above)
Comment (juriczech) [reply packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx]: I thought that's what the `isStakeState` is for and you could use it in the selector. Could you elaborate?

File: suite-native/module-send/src/screens/SendReviewScreen.tsx (hunk 18 above)
Comment (PeKne) [reply suite-native/module-send/src/screens/SendReviewScreen.tsx]: Yes, it does, but Typescript does not know this. I have to assert that the value is present to make the linter happy.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 11 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: I don't agree. The name `useAsyncDebounce` is telling you, that it makes the call "asynchronously debounce" rather than it accepts only the async functions.

File: suite-native/module-send/src/screens/SendReviewScreen.tsx (hunk 18 above)
Comment (juriczech) [reply suite-native/module-send/src/screens/SendReviewScreen.tsx]: Of course, I didn't realise. In that case, I'd consider putting the text and button in a new component. It will be better to have it there because: 
- you can send signedTransaction as prop that is required
- it would accept accountKey, so you will have access to account

I believe that is better and you can keep this function without redundant condition, that is already checked in JSX.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 11 above)
Comment (juriczech) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: I think that's not correct. The async means that the debounce returns a promise. Since it doesn't always return promise anymore, it's not correct and is just redundant. I doubt that it means "asynchronously debounce", since debounce is always like that, that's why it's called `debounce` and not `asynchronous debounce` 😄 I went a little bit far with the metaphor here, but had to prove a point because I think we're looking at this differently. 

Since this is pretty easy to rename (pretty quick if you use the rename symbol in VS code), can we do that to keep this hook name simple and to the point? 

If this was a complicated refactor, it's obviously not too important and could be ignored, but I think we spent more time on this already than if we would just rename it. 

Happy to discuss more if you want though.

File: suite-native/module-send/src/screens/SendReviewScreen.tsx (hunk 18 above)
Comment (PeKne) [reply suite-native/module-send/src/screens/SendReviewScreen.tsx]: Fixed in [affdc58](https://github.com/trezor/trezor-suite/pull/12968/commits/affdc5880efe6a9efaddd99b05b742b2a753ebbb) by moving the button to the separate component.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 11 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx]: fixed 96cbb1d01abbc546dc377addafef8a1deb874148

File: suite-common/wallet-types/src/transaction.ts (hunk 2 above)
Comment (PeKne) [reply suite-common/wallet-types/src/transaction.ts:247]: This can not be moved to the `wallet-core` package, since these types are also needed in `wallet-utils` package.

I still review all the types that are here nad found some that should be elsewhere. Some of theme were moved to wallet-core, so even to the packages/suite (since there will be ever used only by desktop). 

Changes are in commit: [e94fba7](https://github.com/trezor/trezor-suite/pull/12968/commits/e94fba7de6abf55db1824d38b13db5ba85a33a09). I highly encourage to review this commit in isolation.

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx (hunk 1 above)
Comment (PeKne) [reply packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx]: Special selector created here: [05f5b4b](https://github.com/trezor/trezor-suite/pull/12968/commits/05f5b4bafada99e14c5cecad39deaa76eab5f064)

Not much optimistic that this solved anything, but the code looks a bit better at least.

File: suite-native/module-send/src/components/SendOutputsForm.tsx (hunk 15 above)
Comment (PeKne) [reply suite-native/module-send/src/components/SendOutputsForm.tsx:124]: This will be revisited in the error handling follow-up PR. There will be no direct navigation in this callback, but the error state will be set. Then the module will be able to detect this error in a useEffect and handle it according to its context (in this case it will trigger the navigation action).

Issue: https://github.com/trezor/trezor-suite/issues/13125

File: suite-native/navigation/src/components/GoBackIcon.tsx (hunk 19 above)
Comment (PeKne) [reply suite-native/navigation/src/components/GoBackIcon.tsx]: The navigation will be implemented in the useEffect, that will react to change of the send form error state.

Issue created here: https://github.com/trezor/trezor-suite/issues/13125

File: packages/suite/src/hooks/wallet/form/useCompose.ts [hunk 21]
```diff
@@ -5,20 +5,18 @@ import { FeeLevel } from '@trezor/connect';
...
-    ComposeActionContext,
-    SendContextValues,
     PrecomposedTransaction,
     PrecomposedTransactionCardano,
     PrecomposedLevels,
     PrecomposedLevelsCardano,
 } from '@suite-common/wallet-types';
 import { COMPOSE_ERROR_TYPES } from '@suite-common/wallet-constants';
 import { selectSelectedAccount } from 'src/reducers/wallet/selectedAccountReducer';
+import { SendContextValues, UseSendFormState } from '../../../types/wallet/sendForm';
```
Comment (juriczech) [inline packages/suite/src/hooks/wallet/form/useCompose.ts:19]: 'src/types/wallet/sendForm' - please also update in all other usages of this commit

File: suite-common/wallet-types/src/transaction.ts (hunk 2 above)
Comment (juriczech) [reply suite-common/wallet-types/src/transaction.ts:247]: Please fix the imports to `src/types/wallet/sendForm` (comment visible in commit)

File: packages/suite/src/hooks/wallet/form/useCompose.ts (hunk 21 above)
Comment (PeKne) [reply packages/suite/src/hooks/wallet/form/useCompose.ts:19]: ouch, poorly configured auto-import... Fixed: cba8fb0741

**Review by juriczech — APPROVED** [review]

**Comment by PeKne** [comment]
@Nodonisko the `updateFeeInfoThunk` call added ✅
